gprofiler/profilers/java.py

```diff
@@ -789,7 +789,7 @@ def _disable_profiling(self, cause: str) -> None:
             self._safemode_disable_reason = cause
 
     def _profiling_skipped_profile(self, reason: str, comm: str) -> ProfileData:
-        return ProfileData(self._profiling_error_stack("skipped", reason, comm), None, None)
+        return ProfileData(self._profiling_error_stack("skipped", reason, comm), None, None, None)
```

Comment:
This change (generally wherever you pass `None` as container name for a process that we don't profile due to skip / error, but the process is still alive) - because past code would have added the container name in `merge.py`.

Now, you add the name in `perf.py` and in e.g `java.py` for a java process. Then during merging, we select the java `ProfileData`. If the java was `Profiling skipped`, it will have no container name, but the perf one will have!

I suggest: if the process `ProfileData` has no name, but the perf one has, take the perf. This keeps the past behavior for processes we don't profile with a runtime profiler, and gives your enhanced behavior for processes that we do profile with e.g `JavaProfiler`.